src/components/ImageView/ChannelMapView/ChannelMapViewComponent.tsx

```diff
@@ -0,0 +1,410 @@
+import * as React from "react";
+import {NonIdealState} from "@blueprintjs/core";
+import _ from "lodash";
+import {observer} from "mobx-react";
+
+import {CursorInfo, ImageType} from "models";
+import {AppStore, ChannelMapStore, FrameStore} from "stores";
+
+import {BeamProfileOverlayComponent} from "../BeamProfileOverlay/BeamProfileOverlayComponent";
+import {ColorbarComponent} from "../Colorbar/ColorbarComponent";
+import {CursorOverlayComponent} from "../CursorOverlay/CursorOverlayComponent";
+import {OverlayComponent} from "../Overlay/OverlayComponent";
+import {RasterViewComponent} from "../RasterView/RasterViewComponent";
+import {RegionViewComponent} from "../RegionView/RegionViewComponent";
+import {ToolbarComponent} from "../Toolbar/ToolbarComponent";
+
+import {ChannelMapLabelComponent} from "./ChannelMapLabelComponent";
+
+export class ChannelMapViewComponentProps {
+    frame: FrameStore;
+    docked: boolean;
+    channelMapStore: ChannelMapStore;
+    renderWidth: number;
+    renderHeight: number;
```

Comment:
Perhaps we don’t need to set `channelMapStore`, `renderWidth`, and `renderHeight` as props since we always use `AppStore.Instance.channelMapStore` and `AppStore.Instance.overlayStore`.